### ISSUE TITLE
Bots don't replace O2 bottles and just suffocate

### ISSUE THREAD

**Issue by ThyRavenXIII** [Regalis11/Barotrauma#11652]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Vanilla game, no mods, fresh campaign. Bot will use a diving suit/mask when on the sub, but when it runs out of oxygen, even if the bot has O2 in its inventory or nearby they will instead just suffocate to death rather than swap them out.

### Reproduction steps

1. Start a multiplayer campaign.
2. Make sure the Bot has O2 Bottles in its inventory.
3. Give the Bot an order (like operate reactor).
4. Cause a hull breach elsewhere in the ship and don't assign them to fix it.
5. Wait till the ship floods.
6. Let the bot apply mask / fetch dive suit.
7. Wait, after the initial O2 bottle runs out, the bot will suffocate.
8. Inspect inventory and can still see unused O2 bottles after bot death.

### Bug prevalence

Happens every time I play

### Version

v1.0.8.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
N/A
```

**Comment by CurePindal** [comment]
Heres the Steam thread this was discussed
https://steamcommunity.com/app/602960/discussions/0/3818529563380061515/

**Comment by ThyRavenXIII** [comment]
Second Thread https://steamcommunity.com/app/602960/discussions/0/3818529563389115885/

**Comment by itchyOwl** [comment]
Thanks for the report(s), we'll look into fixing it soon!

**Comment by war3scv** [comment]
it seem the diving mask is the source of the problem, if BOT have diving mask in their inventory then they will never swap oxygen tank or remove diving suit.

**Comment by dowdym** [comment]
Bots will not stop tasks to replace oxygen tanks. All turret operators they will stay in the turret until they pass out regardless of how many oxygen tanks they have in their inventory. Doctors will start performing CPR then pass out because they ran out of air and people doing mechanical and electrical repairs will have the same problem.

**Comment by itchyOwl** [comment]
Can't repro after Regalis11/Barotrauma-developmemt@6686d2a, so it may have been simple as that.

> it seem the diving mask is the source of the problem, if BOT have diving mask in their inventory then they will never swap oxygen tank or remove diving suit.

There's something wrong with the logic, because irc bots shouldn't equip masks when they have a suit, yet they seem to do that. Yet seems to be unrelated to the suffocation issue.

**Comment by itchyOwl** [comment]
Fixed the mask issue in Regalis11/Barotrauma-development@a8979dc.

**Comment by gustavomcardoso** [comment]
Hi! is this fixed? Do I need to start a new campaign? 

I just ran into this issue on single player, just had my hole crew die suffocated after a giant spineling encounter that completely destroyed the vessel. They had tool belts with at least 8 O2 tanks in them, they did not have a mask (I have suits for every one and they went directly for the suits because the ship insta flooded) after all was fixed I was getting the pumps back on, and noticed I was playing alone..... checked the bodies, all suffocated to death.... Tragic, had to hit retry option to start again.
